### PR TITLE
[sdk-doc] Doc for SDK overview and SDK validation

### PR DESCRIPTION
**Pull request by jjleng**
### Description
Doc for SDK overview. It explains the layers and components for Aptos SDKs. The doc also presents the test vector validation method we used to validate our SDKs. 

### Test Plan
Previewed locally

---
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/1517)

### CODE REVIEW

File: developer-docs-site/docs/sdks/sdk-overview.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+The Aptos Client component also supports two transaction signing and submission methods.
+
+- Submitting transactions in JSON format, which delegates the signing message (input to the signing function) creation to the
+  API server.
+- Submitting transactions in BCS format, which prepares and signs the raw transactions on the client-side. This method leverages the BCS Library and Transaction Builder for constructing the transaction payloads.
+
+:::notice
+The second method is the recommended way for submitting transactions to the Aptos chain.
+:::
```
Comment (rajkaramchedu) [inline developer-docs-site/docs/sdks/sdk-overview.md:90]: Inserting a new line after `:::notice` and above `:::` will fix the rendering.

File: developer-docs-site/docs/sdks/sdk-overview.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+The Aptos SDKs provide two levels of validation for Transaction Builder and BCS.
+
+1. First, with the unit tests and end-to-end (e2e) tests.
+
+:::notice
+An example of unit tests for the BCS serializer can be found [here](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/transaction_builder/bcs/serializer.test.ts)
+
+An example of an e2e test for submitting a BCS transaction can be found [here](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/aptos_client.test.ts#L88)
+:::
+
```
Comment (rajkaramchedu) [inline developer-docs-site/docs/sdks/sdk-overview.md:111]: On links, like 107 and 109 and everywhere else, always put a period either after "here" like `[here.]` or after the closing parenthesis of the url, like `.../bcs/serializer.test.ts).`

**Comment by rajkaramchedu** [comment]
On the categorization and the filename. In the new docs category SDK, I am anticipating that in addition to the TS SDK, we will also have Python and Rust SDKs, at some point. So we can:

1. Write the SDK overview doc as a general document, as we have it now. This doc now only contains TS SDK details but we can add overview sections for Python and Rust into this same overview doc. In this case, the title Aptos SDK is correct.
2. Or, we can make this overview doc specific to TS only, hence make the filename as `ts-sdk-overview`, change the title to Typescript SDK. I've been thinking about this and I like this second option better. It keeps the docs specific and limited only to the scope of the title and filename. What do you think, @jjleng?

**Comment by jjleng** [comment]
> 

Right now, we only have TS SDK fully maintained. Python SDK is partially maintained. To not make things too complicated, can we do option 1 for now? We can definitely split the doc later if needed.

**Comment by rajkaramchedu** [comment]
> > 
> 
> Right now, we only have TS SDK fully maintained. Python SDK is partially maintained. To not make things too complicated, can we do option 1 for now? We can definitely split the doc later if needed.

Yes. Thanks.

File: developer-docs-site/docs/sdks/sdk-overview.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+:::notice
+
+An example of unit tests for the BCS serializer can be found [here](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/transaction_builder/bcs/serializer.test.ts).
+
+An example of an e2e test for submitting a BCS transaction can be found [here](https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk/src/aptos_client.test.ts#L88).
+
+:::
+
+2. The second level of validation is fuzzing tests with test vectors. The test vectors are produced by the same code used by the Aptos Blockchain. The test vectors are arrays of JSON objects. Each JSON object contains randomized inputs and the expected outputs. These test vectors can be parsed and loaded by Aptos SDKs to validate their implementations of Transaction Builder and BCS.
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/sdk-overview.md:116]: fuzzing? I think it is more ensuring consistency with test vectors

File: developer-docs-site/docs/sdks/sdk-overview.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Vector items are self-explanatory. However, special serialization method is used to save space and avoid data overflow.
+Details are described below:
+
+- All account address are hex-coded.
+- `args` in ScriptFunction is hex-coded.
+- U64 and U128 numbers are serialized as string literals to avoid data truncation.
+- U8 is serialized as a number (not a string).
+- `code` in Script and ModuleBundle are hex-coded.
+
+See how Typescript SDK does vector validation [here](https://github.com/aptos-labs/aptos-core/pull/1461).
```
Comment (davidiw) [inline developer-docs-site/docs/sdks/sdk-overview.md:133]: probably good to point to a specific commit not a PR

**Review by davidiw — CHANGES_REQUESTED** [review]
not sure what happened here, this isn't a reviewable PR and its been here for a while... gonna close it out and leave it to you to resurrect.

**Pull request closed without merging**